controllers/imageclusterinstall_controller.go

```diff
@@ -742,10 +717,14 @@ func (r *ImageClusterInstallReconciler) setClusterInstallMetadata(ctx context.Co
 		return fmt.Errorf("No cluster info found for ImageClusterInstall %s/%s", ici.Namespace, ici.Name)
 	}
 
+	kubeconfigSecret := credentials.KubeconfigSecretName(clusterDDeploymentName)
+	kubeadminPasswordSecret := credentials.KubeadminPasswordSecretName(clusterDDeploymentName)
 	if ici.Spec.ClusterMetadata != nil &&
 		ici.Spec.ClusterMetadata.ClusterID == clusterInfo.ClusterID &&
 		ici.Spec.ClusterMetadata.InfraID == clusterInfo.InfraID &&
-		ici.Spec.ClusterMetadata.AdminKubeconfigSecretRef.Name == kubeconfigSecret {
+		ici.Spec.ClusterMetadata.AdminKubeconfigSecretRef.Name == kubeconfigSecret &&
+		ici.Spec.ClusterMetadata.AdminPasswordSecretRef.Name == kubeadminPasswordSecret {
```

Comment:
I think this needs to be set below (line 737-ish) as you did in `setClusterDeploymentMetadata`